### PR TITLE
feat: Add mapping asyncfunction to method in elementKind

### PR DESCRIPTION
**Pull request by lfvdavid**
Mapping asyncfunction to method

### CODE REVIEW

**Comment by lfvjimisola** [comment]
Please update the test or test resources accordingly.

**Comment by lfvdavid** [comment]
Fixed

File: tests/unit/reqstool_decorators/processors/test_processors.py [hunk 1]
```diff
@@ -25,8 +25,10 @@ def test_get_functions_and_classes(process_decorator_instance: DecoratorProcesso
 
 
 def test_map_type_known_type(process_decorator_instance: DecoratorProcessor):
-    result = process_decorator_instance.map_type("FUNCTION")
-    assert result == "METHOD"
+    map_funcion = process_decorator_instance.map_type("FUNCTION")
+    map_asyncfunction = process_decorator_instance.map_type("ASYNCFUNCTION")
```
Comment (lfvjimisola) [inline tests/unit/reqstool_decorators/processors/test_processors.py:29]: You're just checking the mapping here. Which isn't a sufficient test here.

You need to add all the different types of annotations locations in https://github.com/luftfartsverket/reqstool-python-decorators/blob/19-map-asyncfunction-to-method-in-the-processor/tests/resources/test_decorators/requirements_decorators.py

Comment (lfvjimisola) [reply tests/unit/reqstool_decorators/processors/test_processors.py:29]: @lfvdavid Does the test source code and test class cover all possible decorator locations now?

Comment (lfvdavid) [reply tests/unit/reqstool_decorators/processors/test_processors.py:29]: There is another test under called "test_map_type_unknown_type" that covers "CLASS".

As class is a valid type, and only function and asyncfunction is invalid only those are mapped.

**Review by lfvjimisola — APPROVED** [review]

**Pull request merged**